jest.config.js

```diff
@@ -0,0 +1,13 @@
+/** @type {import('ts-jest').JestConfigWithTsJest} */
+module.exports = {
+  preset: 'ts-jest/presets/js-with-ts', // TODO: change to ts-jest once all repo is converted
+  testEnvironment: 'jsdom',
+  collectCoverageFrom: ['src/**'],
+  resetMocks: true,
+  restoreMocks: true,
+  setupFiles: ['<rootDir>/test/setup.ts'],
+  testPathIgnorePatterns: ['/node_modules/', '<rootDir>/types/'],
+  transform: {
+    '^.+\\.tsx?$': ['ts-jest', { tsconfig: './tsconfig.json' }],
```

Comment:
does ts-jest offer more over using babel-jest?

Reply:
not really, picked ts-jest mainly cause i'm more familiar with it and it has a out-of-the-box preset for ts-with-js setup
https://jestjs.io/docs/getting-started#using-typescript
> Because TypeScript support in Babel is purely transpilation, Jest will not type-check your tests as they are run. If you want that, you can use [ts-jest](https://github.com/kulshekhar/ts-jest) instead, or just run the TypeScript compiler [tsc](https://www.typescriptlang.org/docs/handbook/compiler-options.html) separately (or as part of your build process).

Reply:
i don't have strong opinions other than its nice to compile stuff the same way for tests as when building?

Reply:
yeah i agree - converted to babel!